File: public/dsa/two_pointers/maximum_k_subarray_sum/naive/maximum_k_subarray_sum.cpp
#include <bits/stdc++.h>
using namespace std;

int maxKSubarraySum(vector<int> A, int k) {
    int n = A.size();
    int max_window = 0;
    for (int i = 0; i < n - k + 1; i++) {
        int window = 0;
        for (int j = i; j < i + k; j++) {
            window += A[j];
        }
        max_window = max(max_window, window);
    }
    return max_window;
}

// tests-start
int main() {
    vector<int> A = {3, 5, 6, 2, 4, 7, 8};
    int k = 3;
    cout << maxKSubarraySum(A, k) << endl;
    return 0;
}
// tests-end

/*output
19
*/

File: public/dsa/two_pointers/maximum_k_subarray_sum/naive/maximum_k_subarray_sum.py
from typing import List


class Solution:
    def maxKSubarraySum(self, A: List[int], k: int) -> int:
        n = len(A)
        max_window = 0
        for i in range(n - k + 1):
            window = 0
            for j in range(i, i + k):
                window += A[j]
            max_window = max(max_window, window)
        return max_window


# tests-start
A = [3, 5, 6, 2, 4, 7, 8]
k = 3
sol = Solution()
print(sol.maxKSubarraySum(A, k))
# tests-end

'''output
19
'''
